sql/src/main/java/io/crate/execution/engine/aggregation/impl/MaximumAggregation.java

```diff
@@ -45,15 +45,26 @@
     private final FunctionInfo info;
 
     public static void register(AggregationImplModule mod) {
-        for (final DataType dataType : DataTypes.PRIMITIVE_TYPES) {
-            FunctionInfo functionInfo = new FunctionInfo(
-                new FunctionIdent(NAME, ImmutableList.of(dataType)), dataType, FunctionInfo.Type.AGGREGATE);
-
-            if (dataType instanceof FixedWidthType) {
-                mod.register(new FixedMaximumAggregation(functionInfo));
-            } else {
-                mod.register(new VariableMaximumAggregation(functionInfo));
-            }
+        for (var supportedType : DataTypes.PRIMITIVE_TYPES) {
+            mod.register(
+                Signature.aggregate(
+                    NAME,
+                    supportedType.getTypeSignature(),
+                    supportedType.getTypeSignature()),
+                args -> {
+                    var arg = args.get(0); // f(x) -> x
+                    var info = new FunctionInfo(
+                        new FunctionIdent(NAME, args),
+                        arg,
+                        FunctionInfo.Type.AGGREGATE
+                    );
+                    if (arg instanceof FixedWidthType) {
+                        return new FixedMaximumAggregation(info);
+                    } else {
+                        return new VariableMaximumAggregation(info);
```

Comment:
minor: We could decide this outside of the factory method to avoid "on-call/on-demand" instanceof checks